src/uct/ib/mlx5/ib_mlx5.inl

```diff
@@ -378,7 +378,7 @@ uct_ib_mlx5_set_ctrl_seg(struct mlx5_wqe_ctrl_seg* ctrl, uint16_t pi,
 #else
     uct_ctrl->opmod_idx_opcode      = (opcode << 24) | (htons(pi) << 8) | opmod;
     uct_ib_mlx5_set_ctrl_qpn_ds(ctrl, qp_num, ds);
-    uct_ctrl->dci_stream_channel_id = htons(dci_channel);
+    uct_ctrl->dci_stream_channel_id = ucs_btons(dci_channel);
```

Comment:
maybe just `htons((uint8_t)dci_channel)`

Reply:
> maybe just `htons((uint8_t)dci_channel)`

doesn't work:
objdump -S modules/libuct_ib.so
```
 uct_ctrl->dci_stream_channel_id = htons((uint8_t)dci_channel);              
5084a:   0f b7 85 24 fd ff ff    movzwl -0x2dc(%rbp),%eax                    
50851:   0f b6 c0                movzbl %al,%eax                             
50854:   89 c7                   mov    %eax,%edi                            
50856:   e8 65 7d fc ff          callq  185c0 <htons@plt>                    
```

Reply:
weird that htons is not even inlined.. maybe missing some optimizations?

Reply:
it seems it depends from optimisation level... I tried to build UCX with -O2 switch and compiler excluded all calls to htons at all